server/app/lib/actions/fusor/deployment/deploy_rhev.rb

```diff
@@ -0,0 +1,226 @@
+
+          deployment_id = input.fetch(:deployment_id)
+
+          Rails.logger.warn "XXX finalize: working with deployment: #{deployment_id}"
+
+          deployment = ::Fusor::Deployment.find(deployment_id)
+
+          Rails.logger.warn "XXX finalize: deployment is a: #{deployment.class}"
+
+          if deployment.rhev_engine_host.nil?
+            Rails.logger.warn "XXX finalize: WE DO NOT HAVE A RHEV ENGINE HOST."
+          else
+            Rails.logger.warn "XXX finalize: managed host? #{deployment.rhev_engine_host.managed?}"
+          end
+          Rails.logger.warn "XXX finalize: deployment name is: #{deployment.name}"
+          Rails.logger.warn "XXX finalize: deployment org is: #{deployment.organization.id}"
+
+          Rails.logger.warn "XXX ================ RHEV Hypervisor piece ===================="
+
+          # let's find the proper hostgroup and associate it with the
+          # hypervisors
+          hypervisor_group = find_hypervisor_hostgroup(deployment)
+
+          Rails.logger.warn "XXX finalize: calling assign_host_to_hostgroup (hypervisor)"
+
+          deployment.discovered_hosts.each do |da_host|
+            success, da_host = assign_host_to_hostgroup(da_host, hypervisor_group)
+
+            Rails.logger.warn "XXX finalize: returned from assign_host_to_hostgroup. #{success}"
+            if da_host
+              Rails.logger.warn "XXX finalize: hypervisor host is NOT null! YAY!"
+            else
+              Rails.logger.warn "XXX finalize: damn what happened to the hypervisor host"
+            end
+          end
+
+          Rails.logger.warn "XXX ================ RHEV Engine piece ===================="
+
+          # let's find the proper hostgroup and associate it with the engine
+          engine_group = find_engine_hostgroup(deployment)
+
+          Rails.logger.warn "XXX finalize: calling assign_host_to_hostgroup (engine)"
+
+          success, host = assign_host_to_hostgroup(deployment.rhev_engine_host, engine_group)
+
+          Rails.logger.warn "XXX returned from assign_host_to_hostgroup. #{success}"
+          if host
+            Rails.logger.warn "XXX engine host is NOT null! YAY!"
+          else
+            Rails.logger.warn "XXX damn what happened to the engine host"
+          end
+
+          Rails.logger.warn "XXX ================ Leaving finalize method ===================="
+        end
+
+        def find_engine_hostgroup(deployment)
+          find_hostgroup(deployment, "RHEV-Engine")
+        end
+
+        def find_hypervisor_hostgroup(deployment)
+          find_hostgroup(deployment, "RHEV-Hypervisor")
+        end
+
+        #
+        # "borrowed" from staypuft:
+        # https://github.com/theforeman/staypuft/blob/master/app/controllers/staypuft/deployments_controller.rb#L158-L216
+        #
+        def assign_host_to_hostgroup(assignee_host, hostgroup)
+          converting_discovered = assignee_host.is_a? Host::Discovered
```

Comment:
If the user hasn't configured an engine on the deployment object, I believe this will generate a nil? error.